sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/UpdateCenterDeserializer.java

```diff
@@ -276,11 +277,16 @@ private void parseReleases(Properties p, Sonar sonar, String pluginKey, Componen
     HashMap<String, Map.Entry<String, Integer>> flavosLabel, boolean isPublicRelease, boolean isArchivedRelease) {
     String[] pluginPublicReleases = getArray(p, pluginKey, key);
     for (String pluginVersion : pluginPublicReleases) {
-      Release release = parseRelease(p, sonar, pluginKey, component, isPublicRelease, isArchivedRelease, pluginVersion, flavosLabel);
-      if (!component.getAllReleases().contains(release)) {
-        component.addRelease(release);
+      Release releaseToAdd = parseRelease(p, sonar, pluginKey, component, isPublicRelease, isArchivedRelease, pluginVersion, flavosLabel);
+      Optional<Release> alreadyExistingRelease = component.getAllReleases().stream()
+        .filter(r -> r.getArtifact().equals(releaseToAdd.getArtifact()))
+        .filter(r -> r.getVersion().equals(releaseToAdd.getVersion()))
```

Comment:
```suggestion
        .filter(r -> r.equals(releaseToAdd))
```
little suggestion since the `equals` method already does that